### PR TITLE
Added the 'Configuring GitHub oAuth for OpenShift infrastructure' sec…

### PR DESCRIPTION
**Pull request by supriya-bharadwaj**
…tion to the User Guides > Version Control chapter.

### What does this PR do?
Adds a new section for *Configuring GitHub oAuth for OpenShift infrastructure* in the *User Guides > Version Control* chapter.
File name: `src/main/pages/che-6/user-guide/version-control.adoc`

### What issues does this PR fix or reference?
https://github.com/eclipse/che-docs/issues/731

Hi @artaleks9 , can you please have a look at this section? Thanks!

### CODE REVIEW

**Comment by artaleks9** [comment]
Looks good for me. 
I would like to add and note that these steps an user need to do when uses deployed CHE6 in the _multiuser mode_ and it can be as Openshift infrastructure, as and Docker.

File: src/main/pages/che-6/user-guide/version-control.adoc [hunk 1]
```diff
@@ -128,6 +128,60 @@ endif::[]
...
+[id="configuring-github-oauth-for-openshift-infrastructure"]
+== Configuring GitHub oAuth for OpenShift infrastructure
+
+.Prerequisites
+
+* A deployed instance of Eclipse Che
+
+.Procedure
+
+OAuth for GitHub allows users to import projects using the SSH addresses (`git@`), push to repositories, and use the pull request panel. The steps listed in this section are performed by an administrative user.
```
Comment (rkratky) [inline src/main/pages/che-6/user-guide/version-control.adoc]: pull request panel -- is it a GUI element?

File: src/main/pages/che-6/user-guide/version-control.adoc [hunk 2]
```diff
@@ -128,6 +128,60 @@ endif::[]
...
+
+To enable automatic SSH key upload to GitHub for users, take the following steps:
+
+. In the *Overview* tab, click the _<application-name>_ to open it.
+
+. Click the Keycloak link to open the *Red Hat Singe Sign-On* window.
```
Comment (rkratky) [inline src/main/pages/che-6/user-guide/version-control.adoc]: => Single

File: src/main/pages/che-6/user-guide/version-control.adoc [hunk 3]
```diff
@@ -128,6 +128,60 @@ endif::[]
...
+
+. Expand the *Configure* tab and then click *Identity Providers*.
+
+. In the *Add provider* drop-down list, click *GitHub*. The *Add identity provider* window displays.
```
Comment (rkratky) [inline src/main/pages/che-6/user-guide/version-control.adoc]: => is displayed

**Review by rkratky — CHANGES_REQUESTED** [review]

File: src/main/pages/che-6/user-guide/version-control.adoc (hunk 1 above)
Comment (artaleks9) [reply src/main/pages/che-6/user-guide/version-control.adoc]: Yes, CHE6 has the UI _Pull Request panel_

Comment (supriya-bharadwaj) [reply src/main/pages/che-6/user-guide/version-control.adoc]: Thanks, @artaleks9 !

**Comment by supriya-bharadwaj** [comment]
Thanks for the review, @artaleks9 and @rkratky . I have incorporated your feedback at https://github.com/eclipse/che-docs/pull/736/commits/64ce64b498d602d22663b1da75607773a85719e4. 

@rkratky , after you and @artaleks9  have seen this and approve of the changes, this is ready to be merged. Thanks!

**Comment by artaleks9** [comment]
Looks good. Thanks for your corrections.

**Review by artaleks9 — APPROVED** [review]

File: src/main/pages/che-6/user-guide/version-control.adoc [hunk 4]
```diff
@@ -18,7 +18,7 @@ The following sections describe how to connect and authenticate users to a remot
 
 Private repositories require a secure SSH connection. In order to connect to Git repositories over SSH, an SSH key pair needs to be generated. SSH keys are saved in user preferences, so the same key can be used in all workspaces.
 
-NOTE: HTTPS Git URLs can only be used to push changes when OAuth authentication is enabled. See link:user-management.html#enabling-authentication-with-social-accounts-and-brokering[Enabling authentication with social accounts and brokering].
+NOTE: HTTPS Git URLs can only be used to push changes when oAuth authentication is enabled. See link:user-management.html#enabling-authentication-with-social-accounts-and-brokering[Enabling authentication with social accounts and brokering].
```
Comment (rkratky) [inline src/main/pages/che-6/user-guide/version-control.adoc]: 'OAuth' was correct! https://oauth.net/

File: src/main/pages/che-6/user-guide/version-control.adoc [hunk 5]
```diff
@@ -88,12 +88,12 @@ image::git/ssh_gitlab.png[]
 
 
 [id="configuring-github-oauth"]
-== Configuring GitHub OAuth
+== Configuring GitHub oAuth
```
Comment (rkratky) [inline src/main/pages/che-6/user-guide/version-control.adoc]: 'OAuth' was correct! https://oauth.net/

File: src/main/pages/che-6/user-guide/version-control.adoc [hunk 6]
```diff
@@ -88,12 +88,12 @@ image::git/ssh_gitlab.png[]
...
 
-OAuth for Github allows users to import projects using SSH addresses (`git@`), push to repositories, and use the pull request panel. To enable automatic SSH key upload to GitHub for users:
+oAuth for Github allows users to import projects using SSH addresses (`git@`), push to repositories, and use the pull request panel. To enable automatic SSH key upload to GitHub for users:
```
Comment (rkratky) [inline src/main/pages/che-6/user-guide/version-control.adoc]: * 'OAuth' was correct! https://oauth.net/
* => `the *Pull Request* panel`

File: src/main/pages/che-6/user-guide/version-control.adoc [hunk 7]
```diff
@@ -126,8 +126,64 @@ ifeval::["{project-context}" == "che"]
 endif::[]
 ====
 
-Once OAuth is configured, SSH keys are generated and uploaded automatically to GitHub by a user in the IDE in *Profile > Preferences > SSH > VCS* by clicking the _Octocat_ icon. You can connect to your GitHub account and choose repositories to clone, rather than having to manually type (or paste) GitHub project URLs.
+Once OAuth is configured, SSH keys are generated and uploaded automatically to GitHub by a user in the IDE in *Profile* > *Preferences* > *SSH* > *VCS* by clicking the _Octocat_ icon. You can connect to your GitHub account and choose repositories to clone, rather than having to manually type (or paste) GitHub project URLs.
 
+[id="configuring-github-oauth-for-openshift-infrastructure"]
+== Configuring GitHub oAuth for OpenShift infrastructure
```
Comment (rkratky) [inline src/main/pages/che-6/user-guide/version-control.adoc]: OAuth

File: src/main/pages/che-6/user-guide/version-control.adoc [hunk 8]
```diff
@@ -126,8 +126,64 @@ ifeval::["{project-context}" == "che"]
...
+
+.Prerequisites
+
+* A deployed instance of Eclipse Che
+
+.Procedure
+
+oAuth for GitHub allows users to import projects using the SSH addresses (`git@`), push to repositories, and use the *Pull Request* panel. The steps listed in this section are performed by an administrative user.
```
Comment (rkratky) [inline src/main/pages/che-6/user-guide/version-control.adoc]: OAuth

File: src/main/pages/che-6/user-guide/version-control.adoc [hunk 9]
```diff
@@ -126,8 +126,64 @@ ifeval::["{project-context}" == "che"]
...
+
+. In the *Add identity provider* window:
+
+.. The *Redirect URI* field, depending on your Eclipse Che installation, type the Authorization callback URL.
+
+.. Copy the client ID and the client secret from GitHub and paste them in the *Client ID* and the *Client Secret* fields, respectively.
+
+.. In the *Homepage URL* and the *Authorization callback URL* field, depending on your Eclipse Che installation, type the two URLS.
+
+.. In the *Default Scopes* field, type _repo, user, write:publick_key_.
```
Comment (rkratky) [inline src/main/pages/che-6/user-guide/version-control.adoc]: =>
```
type `repo, user, write:publick_key`
```

File: src/main/pages/che-6/user-guide/version-control.adoc [hunk 10]
```diff
@@ -126,8 +126,64 @@ ifeval::["{project-context}" == "che"]
...
+
+. Click the *Client Roles* drop-down menu and click *broker*.
+
+. In the *Available Roles* list, click *read-token*.
+
+. Click *Add selected*.
+
+. In the *Effective Roles* list, click *read-token*.
+
+The GitHub oAuth is now set for the user.
```
Comment (rkratky) [inline src/main/pages/che-6/user-guide/version-control.adoc]: OAuth

**Review by rkratky — CHANGES_REQUESTED** [review]

**Comment by rkratky** [comment]
@supriya-bharadwaj, some more. Thanks.

**Review by rkratky — APPROVED** [review]

**Pull request merged**